src/main/java/ch/njol/skript/expressions/ExprCommandInfo.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.command.Command;
+import org.bukkit.command.PluginCommand;
+import org.bukkit.command.defaults.BukkitCommand;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.command.Commands;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+enum InfoType {
+	NAME,
+	DESCRIPTION,
+	LABEL,
+	USAGE,
+	ALIASES,
+	PERMISSION,
+	PERMISSION_MESSAGE,
+	PLUGIN,
+}
+
+@Name("Command Info")
+@Description("Get information about a command.")
+@Examples({"main name of command \"skript\"",
+	"description of command \"help\"",
+	"label of command \"pl\"",
+	"usage of command \"help\"",
+	"aliases of command \"bukkit:help\"",
+	"permission of command \"/op\"",
+	"command \"op\"'s permission message",
+	"command \"sk\"'s plugin owner"})
+@Since("INSERT VERSION")
+public class ExprCommandInfo extends SimpleExpression<String> {
+	
+	static {
+		Skript.registerExpression(ExprCommandInfo.class, String.class, ExpressionType.SIMPLE,
+			"[the] main [command] [name] of command %string%", "command %string%'s main [command] [name]",
+			"[the] description of command %string%", "command %string%'s description",
+			"[the] label of command %string%", "command %string%'s label",
+			"[the] usage of command %string%", "command %string%'s usage",
+			"[the] aliases of command %string%", "command %string%'s aliases",
+			"[the] permission of command %string%", "command %string%'s permission",
+			"[the] permission message of command %string%", "command %string%'s permission message",
+			"[the] plugin [owner] of command %string%", "command %string%'s plugin [owner]");
```

Comment:
```suggestion
			"[the] main command [label] of %strings%", "%strings%'[s] main command [label]",
			"[the] description of command %strings%", "command %strings%'[s] description",
			"[the] label of command %strings%", "command %strings%'[s] label",
			"[the] usage of command %strings%", "command %strings%'[s] usage",
			"[the] aliases of command %strings%", "command %strings%'[s] aliases",
			"[the] permission of command %strings%", "command %strings%'[s] permission",
			"[the] permission message of command %strings%", "command %strings%'[s] permission message",
			"[the] plugin [owner] of command %strings%", "command %strings%'[s] plugin [owner]");
```
The code should be edited accordingly to support multiple commands

Reply:
I don't think people would use them to get multiple commands, but it's up for discussion

Reply:
This wasn't resolved, and believe me people find ways to make use of the expression returning multiple commands so it should be supported. I provided a class that should help with it on Discord, if anything.

Reply:
Also make sure to change the isSingle method when adding multiple value support